packages/api/src/routes/medical/document.ts

```diff
@@ -123,6 +124,11 @@ router.post(
     const cxDocumentRequestMetadata = cxRequestMetadataSchema.parse(req.body);
     const forceCommonwell = stringToBoolean(getFrom("query").optional("commonwell", req));
     const forceCarequality = stringToBoolean(getFrom("query").optional("carequality", req));
+    const optingOut = await isPatientOptingOut({ cxId, patientId });
+
+    if (optingOut) {
+      throw new ForbiddenError("Patient has opted out of data sharing");
```

Comment:
This is important logic that we want to make sure its in place not only on the route level, but on the command level. I think we should move it there.

Also, 403/Forbidden is specific to access control. We can throw BadRequestError/400 or some other 4xx error code - like 422, but I think 400 with a custom message is good enough.